roofit/roofitcore/src/RooClassFactory.cxx

```diff
@@ -426,27 +426,25 @@ bool RooClassFactory::makeClass(const char* baseName, const char* className, con
 
   if (realArgNames && *realArgNames) {
     const size_t bufSize = strlen(realArgNames)+1;
-    char* buf = new char[bufSize] ;
-    strlcpy(buf,realArgNames,bufSize) ;
-    char* token = strtok(buf,",") ;
+    std::vector<char> buf(bufSize);
```

Comment:
Is it not better using direct std::string instead of std::vector<char> ?

Reply:
Good idea :+1:! I will update that. It could also be faster because of the small string optimization.

Reply:
Actually replacing the algorithms with `std::string`-based things is not so easy. There are some places where I don't understand the existing logic and where I think it's incorrect.

I'll merge this PR as it is, maybe I'll follow up on the `std::string` thing with a GitHub issue so I won't forget.